libraries/Microsoft.Bot.Builder.Dialogs.Adaptive/AdaptiveDialog.cs

```diff
@@ -674,7 +674,7 @@ protected async Task<DialogTurnResult> ContinueActionsAsync(DialogContext dc, ob
                 else
                 {
                     // Set interrupted flag
-                    if (interrupted && !actionDC.State.ContainsKey(TurnPath.Interrupted))
+                    if (interrupted && !actionDC.State.TryGetValue(TurnPath.Interrupted, out _))
```

Comment:
The fact that there is different semantics between these lines is disturbing. Note ContainsKey and TryGetValue are both part of the IDictionary contract they shouldn't have such differing semantics.